src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/msg/Msg.java

```diff
@@ -8,5 +11,13 @@
     byte[] getMsgBody();
 
     boolean isTestOnly();
-    
+
+    String getCaseNumber();
+
+    Classification getClassification();
+
+    String getJurisdiction();
+
+    List<String> getDocumentUrls();
+
 }
```

Comment:
These changes defeat the purpose of having a common interface among messages. The new fields should only be in EnvelopeMsg and getMsgBody() should take care of placing them in the byte[].

Reply:
@luigibk Sure, I'm assuming the receiving end will take care of unmarshalling the message to the correct subtype.

Reply:
@aliveni yes that is right. What I am saying is different. The serviceBusHelper encapsulates (and isolates) the Azure-ServiceBus client dependency. The send and sendAsync methods there take a Msg, which is a generic interface that can be converted to a BusMessage, the Azure client representation of a message. If you introduce all those fields in the Msg interface, instead of just a body, all the implementors of the Msg interface, going forward, will have to have those fields, which is a rather strict requirement. What I would suggest is that the fields are only present in EnvelopeMsg and in that class you implement Msg#getBody(), which will convert those fields in the message body.